vault/resource_ldap_secret_backend_test.go

```diff
@@ -62,51 +62,132 @@ func TestLDAPSecretBackend(t *testing.T) {
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, ""),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDefaultLeaseTTL, "0"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldMaxLeaseTTL, "0"),
 					resource.TestCheckResourceAttr(resourceName, consts.FieldBindDN, bindDN),
 					resource.TestCheckResourceAttr(resourceName, consts.FieldBindPass, bindPass),
 					resource.TestCheckResourceAttr(resourceName, consts.FieldURL, url),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldUserDN, updatedUserDN),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldInsecureTLS, "false"),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldConnectionTimeout, "99"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldUserDN, userDN),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldInsecureTLS, "true"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldConnectionTimeout, "30"),
+				),
+			},
+			{
+				Config: testLDAPSecretBackendConfig_ad(path, url, `description = "new description"`),
+				Check: resource.ComposeTestCheckFunc(
+					func(*terraform.State) error {
+						client := testProvider.Meta().(*provider.ProviderMeta).GetClient()
+						if _, err := client.Logical().Write(path+"/rotate-root", nil); err != nil {
+							return err
+						}
+						return nil
+					},
+					resource.TestCheckResourceAttr(resourceName, consts.FieldSchema, "ad"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, "new description"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDefaultLeaseTTL, "0"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldMaxLeaseTTL, "0"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBindDN, bindDN),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldURL, url),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldUserDN, userDN),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldInsecureTLS, "true"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldConnectionTimeout, "30"),
+
+					// Even though we did a rotate-root, we expect the TF state
+					// to have no change for bindpass because we did not
+					// explicitly change it in the config and it is not
+					// returned by the Vault API.
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBindPass, bindPass),
+				),
+			},
+			{
+				Config: testLDAPSecretBackendConfig_ad_updated(path, url, `description = "new description"`),
+				Check: resource.ComposeTestCheckFunc(
+					// We explicitly updated the bindpass in the TF config so
+					// we expect the state to contain the new value.
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBindPass, "NEW-SuperSecretPassw0rd"),
```

Comment:
Since the bindpass is now being updated and sent to Vault, is this test step expected to result in an error since the updated password we send and the password in Vault (generated by rotate root) don't match? In which case should we add a `ExpectError` here?

Reply:
Ah I see your comment on the DB engine fix [PR here](https://github.com/hashicorp/terraform-provider-vault/pull/1876/files#r1220324290). Does this happen to work because of how LDAP is set up?

Reply:
Yes, I believe it is a feature of both LDAP and the LDAP engine. The engine will fallback to the previous password for up to 10 minutes.

I was actually thinking about it this morning. This is out of scope for this PR, but I wonder if the LDAP engine should prevent updating the bindpass after a rotate-root operation?

Reply:
Ah interesting, I see what you mean. It definitely feels like very specific behavior that we should at least document for the LDAP engine if we don't end up looking into changing the behavior